From 887dac30f061155ba36ca007136016061dd88df1 Mon Sep 17 00:00:00 2001
From: Faouaz TENOUTIT <faouaz.tenoutit@intel.com>
Date: Mon, 25 Aug 2014 21:16:24 +0200
Subject: [PATCH] sound: sst: Make DSP firmware download on BYT optional

BYT core can do efficient and reliable soft decoding for MP3
and AAC. It can also do HW decoding iff the right codec firmwares
are installed.
This patch allows selection between SW and HW decoding, mostly
to avoid relying on alpha codec firmwares.

Change-Id: I62f295f1ac881bbb11580e4b67e78dfcba1de933
Signed-off-by: Faouaz TENOUTIT <faouaz.tenoutit@intel.com>
---
 sound/soc/intel/Kconfig        | 10 ++++++++++
 sound/soc/intel/sst/sst.c      |  1 +
 sound/soc/intel/sst/sst_acpi.c |  4 ++++
 3 files changed, 15 insertions(+)

diff --git a/sound/soc/intel/Kconfig b/sound/soc/intel/Kconfig
index aea6bfd..9d6cf46 100644
--- a/sound/soc/intel/Kconfig
+++ b/sound/soc/intel/Kconfig
@@ -54,3 +54,13 @@ config SST_DPCM
 config SND_SST_MACHINE
 	tristate
 
+config SND_SST_SW_CODECS
+	depends on SND_INTEL_SST
+	bool "Disable HW codecs"
+	default n
+
+	help
+		This disables HW codecs support and falls back on
+		SW decoding. This is useful when HW codec firmwares
+		are still at alpha quality.
+
diff --git a/sound/soc/intel/sst/sst.c b/sound/soc/intel/sst/sst.c
index 5a7d992..2ca6eff 100644
--- a/sound/soc/intel/sst/sst.c
+++ b/sound/soc/intel/sst/sst.c
@@ -433,6 +433,7 @@ int sst_driver_ops(struct intel_sst_drv *sst)
 		sst->tstamp = SST_TIME_STAMP_MRFLD;
 		sst->ops = &mrfld_ops;
 		sst->ops->do_recovery = sst_do_recovery;
+		sst->ops->post_download = sst_post_download_byt;
 #endif
 
 		return 0;
diff --git a/sound/soc/intel/sst/sst_acpi.c b/sound/soc/intel/sst/sst_acpi.c
index 6074b33..6ad38dc 100644
--- a/sound/soc/intel/sst/sst_acpi.c
+++ b/sound/soc/intel/sst/sst_acpi.c
@@ -196,7 +196,11 @@ static const struct sst_lib_dnld_info  byt_lib_dnld_info = {
 	.mod_end            = SST_BYT_IMR_VIRT_END,
 	.mod_table_offset   = BYT_FW_MOD_TABLE_OFFSET,
 	.mod_table_size     = BYT_FW_MOD_TABLE_SIZE,
+#ifndef CONFIG_SND_SST_SW_CODECS
 	.mod_ddr_dnld       = true,
+#else
+	.mod_ddr_dnld       = false,
+#endif
 };
 
 static const struct sst_ipc_info cht_ipc_info = {
-- 
1.8.3.2

